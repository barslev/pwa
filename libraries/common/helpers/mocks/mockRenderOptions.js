import PropTypes from 'prop-types';

const options = {
  context: {
    i18n: () => ({
      // eslint-disable-next-line no-underscore-dangle
      __(input) {
        if (input) {
          return input;
        }
        return '';
      },
      _p: () => 'p',
      _d: () => 'd',
      _n: () => 'n',
    }),
  },
  childContextTypes: {
    i18n: PropTypes.func,
  },
};

export default options;
