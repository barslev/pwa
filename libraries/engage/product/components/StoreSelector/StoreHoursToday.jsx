import React from 'react';
import PropTypes from 'prop-types';
import I18n from '@shopgate/pwa-common/components/I18n';
import { storeHoursToday } from './style';

const weekdays = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'];

/**
 * Renders the store's opening hours for "today".
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function StoreHoursToday({ hours }) {
  if (!hours) {
    return null;
  }

  const today = weekdays[new Date().getDay()];
  const hoursToday = hours[today] || null;

  if (!hoursToday || hoursToday === '') {
    return null;
  }

  return (
    <div className={storeHoursToday}>
      <I18n.Text string="product.location.today_hours" params={{ hours: hoursToday }} />
    </div>
  );
}

StoreHoursToday.propTypes = {
  hours: PropTypes.shape(),
};

StoreHoursToday.defaultProps = {
  hours: null,
};

export default StoreHoursToday;
