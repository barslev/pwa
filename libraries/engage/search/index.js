/** @module search */

// ACTIONS
export { default as fetchSearchResults } from '@shopgate/pwa-common-commerce/search/actions/fetchSearchResults';
export { default as fetchSearchSuggestions } from '@shopgate/pwa-common-commerce/search/actions/fetchSearchSuggestions';

// CONSTANTS
export * from '@shopgate/pwa-common-commerce/search/constants/index';
export * from '@shopgate/pwa-common-commerce/search/constants/Pipelines';
export * from '@shopgate/pwa-common-commerce/search/constants/Portals';

// HELPERS
export { default as removeHighlightingPlaceholers } from '@shopgate/pwa-common-commerce/search/helpers/removeHighlightingPlaceholders';
export * from '@shopgate/pwa-common-commerce/search/helpers';

// SELECTORS
export * from '@shopgate/pwa-common-commerce/search/selectors';

// STREAMS
export * from '@shopgate/pwa-common-commerce/search/streams';
