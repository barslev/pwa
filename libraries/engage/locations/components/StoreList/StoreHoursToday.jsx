import React from 'react';
import PropTypes from 'prop-types';
import { i18n } from '@shopgate/engage/core';
import { storeHoursToday } from './style';

const weekdays = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'];

/**
 * Renders the store's opening hours for "today".
 * @param {Object} props The component props.
 * @returns {JSX}
 */
function StoreHoursToday({ hours }) {
  if (!hours) {
    return null;
  }

  const today = weekdays[new Date().getDay()];
  const hoursToday = hours[today] || null;

  if (!hoursToday || hoursToday === '') {
    return null;
  }

  return (
    <div className={storeHoursToday}>
      {i18n.text('product.location.today_hours', { hours: hoursToday })}
    </div>
  );
}

StoreHoursToday.propTypes = {
  hours: PropTypes.shape(),
};

StoreHoursToday.defaultProps = {
  hours: null,
};

export default StoreHoursToday;
