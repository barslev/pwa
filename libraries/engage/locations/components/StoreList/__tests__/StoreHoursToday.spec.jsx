import React from 'react';
import { shallow } from 'enzyme';
import { mockDate, resetMockDate } from '@shopgate/engage/mocks';
import StoreHoursToday from '../StoreHoursToday';

const hours = {
  mon: '8:00am - 4:00pm',
  tue: '8:00am - 4:00pm',
  wed: '8:00am - 8:00pm',
  thu: '8:00am - 8:00pm',
  fri: '8:00am - 8:00pm',
  sat: '8:00am - 8:00pm',
  sun: '8:00am - 8:00pm',
};

describe('<StoreHoursToday />', () => {
  afterEach(() => {
    resetMockDate();
  });

  it('should not render if no hours have been passed', () => {
    const wrapper = shallow(<StoreHoursToday />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance()).toEqual(null);
  });

  it('should render as expected', () => {
    mockDate('2019-01-01T10:00:00.000Z');
    const wrapper = shallow(<StoreHoursToday hours={hours} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div').getElements().length).toEqual(1);
  });
});
