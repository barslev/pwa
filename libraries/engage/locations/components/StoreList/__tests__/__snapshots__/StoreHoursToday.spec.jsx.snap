// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreHoursToday /> should not render if no hours have been passed 1`] = `""`;

exports[`<StoreHoursToday /> should render as expected 1`] = `
<div
  className={
    Object {
      "data-css-1kqiarg": "",
    }
  }
>
  product.location.today_hours
</div>
`;
